### DISCUSSION TITLE
Pagination does not scroll to top

### DISCUSSION THREAD

**Discussion by fonz-lu** [livewire/livewire#7188] [1. Help]
Hi,

according to the documtation, pagination should scroll to top by default, but it isn't in my app.

I tried adding parameters, but with no effect {{ $trainings->links(data: ['scrollTo' => true, 'scrollTo' => '#paginated-trainings']) }}

Pagination is working and there no errors in the console. I am on v 3.0.10

Thanks

**Comment by vorban** [comment]
@fonz-lu I've got the exact same problem, with a custom pagination view. You didn't specify if you were using a custom view or not, but if you do, this may be of interest to you:

The `scrollTo` behaviour is defined **in the blade views** of the `Features/SupportsPagination`, not in the trait `WithPagination`. Check out [this line](https://github.com/livewire/livewire/blob/7c1f609515e74ef1197c08e56a5606571b3ec1d9/src/Features/SupportPagination/views/simple-tailwind.blade.php#L24):
```html
<!-- replaced irrelevant parts with "..." -->
<button ... wire:click="setPage(...)" x-on:click="{{ $scrollIntoViewJsSnippet }}" ...>
```


The `scrollIntoViewJsSnippet` is defined [at the top of the file](https://github.com/livewire/livewire/blob/7c1f609515e74ef1197c08e56a5606571b3ec1d9/src/Features/SupportPagination/views/simple-tailwind.blade.php#L1):
```blade
@php
if (! isset($scrollTo)) {
    $scrollTo = 'body';
}

$scrollIntoViewJsSnippet = ($scrollTo !== false)
    ? <<<JS
       (\$el.closest('{$scrollTo}') || document.querySelector('{$scrollTo}')).scrollIntoView()
    JS
    : '';
@endphp
``` 

As a solution you can copy paste this snippet and call it in your pagination elements.
For my purpose a simpler implementation will suffice.

Since v3 allows us to[ return custom JS to be executed](https://livewire.laravel.com/docs/actions#evaluating-one-off-javascript-expressions), I would have though the `WithPagination` trait used this feature to trigger a scrollTo on page change, especially since the *scroll or not to scroll* check itself is already in PHP.

I hope this solves your issue.
